### PR TITLE
Update to NAN 2.0

### PR DESCRIPTION
**Pull request by kkoopa**

### CODE REVIEW

**Comment by TooTallNate** [comment]
Thanks for this!

File: test/callback.js [hunk 1]
```diff
@@ -22,13 +22,6 @@ describe('weak()', function () {
       assert(called)
     })
 
-    it('should pass the target in as the first argument', function () {
-      weak({ foo: 'bar' }, function (o) {
```
Comment (TooTallNate) [inline test/callback.js]: Can you explain the API change please?

Comment (kkoopa) [inline test/callback.js]: The persistent gets deallocated before the weak callback executes, it is not allowed to access it.

**Comment by TooTallNate** [comment]
@kkoopa Any idea what's up with iojs v3 https://travis-ci.org/TooTallNate/node-weak/jobs/76006053 ?

**Comment by kkoopa** [comment]
C++11 requires g++-4.8. Check NAN's appveyor conf.

**Comment by TooTallNate** [comment]
@kkoopa Do you know what the recommended fix for that in Travis is at the moment?

**Comment by kkoopa** [comment]
Yes, as I said, [add this to .travis.yml](https://github.com/nodejs/nan/blob/master/.travis.yml#L3-L10):
```yaml
env:
  - CXX=g++-4.8
addons:
  apt:
    sources:
    - ubuntu-toolchain-r-test
    packages:
    - g++-4.8
```

**Comment by TooTallNate** [comment]
Oh sorry, I missed the __NAN's__ part there :p Will update.

**Pull request merged**